### PR TITLE
[style] consistent nn. and nn.functional

### PR DESCRIPTION
**Pull request by stas00**
As discussed in https://github.com/huggingface/transformers/issues/11600 this PR normalizes to `nn.functional.foo()` replacing `F.` and `torch.nn.` with `nn.`.

This is all automated by:
```
# deal with torch.nn
perl -pi -e 's|^import torch\n|from torch import nn\nimport torch\n|' `grep -Ilr torch.nn src`
find src -type f -exec perl -X -pi -e 's{(?<!(from |import |[`#/]))torch\.nn\.}{nn.}g' {} \;
find src -type f -exec perl -pi -e 's|import torch\.nn as nn|from torch import nn|g' {} \;

# deal with F
find src -type f -exec perl -pi -e 's|from torch.nn import functional as F|from torch import nn|g' {} \;
find src -type f -exec perl -pi -e 's|import torch.nn.functional as F|from torch import nn|g' {} \;
find src -type f -exec perl -pi -e 's|(?<!\w)F\.|nn.functional.|g' {} \;

git checkout src/transformers/data/data_collator.py
perl -pi -e 's|import torch||' src/transformers/models/prophetnet/convert_prophetnet_original_pytorch_checkpoint_to_pytorch.py
make fixup
```

This is just `src/transformers` for now. If happy I can do the same for `templates`, `tests` and `examples` next

To kind reviewers: This is a massive auto-rewrite, if you notice some missed patterns please just point one instance to me, so I will adjust regex to catch them all. (since we need to do at least `tests`/`examples` too)

@sgugger, @LysandreJik, @patrickvonplaten

### CODE REVIEW

**Review by patrickvonplaten — APPROVED** [review]

**Review by sgugger — APPROVED** [review]
Thanks a lot for all the PERL magic!
